### PR TITLE
Add arm? helper to chef-utils

### PR DESCRIPTION
**Pull request by tas50**
It's tagged as first shipping in chef 15.10 since I plan to backport this and that'll make sense in the docs.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: chef-utils/README.md [hunk 1]
```diff
@@ -108,17 +108,21 @@ The OS helpers provide an alternative to comparing data from `node['os']`.
 
 Architecture Helpers allow you to determine the processor architecture of your node.
 
-* `_64_bit?`
 * `_32_bit?`
+* `_64_bit?`
+* `aarch64?`
+* `arch64?`
+* `arm64?`
```
Comment (lamont-granquist) [inline chef-utils/README.md:115]: these helpers don't actually exist (as the specs show)

Comment (tas50) [reply chef-utils/README.md]: Updated

File: chef-utils/lib/chef-utils/dsl/architecture.rb [hunk 2]
```diff
@@ -103,6 +103,17 @@ def powerpc?(node = __getnode)
...
 
+      # Determine if the current architecture is arm
+      #
+      # @since 15.10
+      #
+      # @return [Boolean]
+      #
+      def arm?(node = __getnode)
+        %w{armhf aarch64 arm64 arch64}
+        .include?(node["kernel"]["machine"])
```
Comment (phiggins) [inline chef-utils/lib/chef-utils/dsl/architecture.rb:114]: Super nitpick but using two lines like this is kind of a weird way to write this statement.

Comment (tas50) [reply chef-utils/lib/chef-utils/dsl/architecture.rb]: True. Fixed

**Pull request merged**